### ISSUE TITLE
quickstart OpenAI fails

### ISSUE THREAD

**Issue by distributev** [cpacker/MemGPT#1103]
This is about 

https://github.com/cpacker/MemGPT/issues/1077

It is still not working and I did further investigations.

1. It gives the same error on both Windows and Linux (initially I tried on linux)
2. From the same computer I am able to access https://openai.com in the browser without any issue
3.  I did a test and I purposely provided an invalid api key inside .memgpt/credentials file to see if anything changes and I got "unauthorized" - getting "unauthorized" means that there are no network anomalies
![image](https://github.com/cpacker/MemGPT/assets/19224635/cb2aa319-5b44-4b7a-a83f-fb7e3798415f)
4. using curl I am able to access the same endpoint - in curl I give correct auth key it works and when I give invalid auth key to curl I get unauthorized

The network is working.

**Comment by distributev** [comment]
Is it possible for the displayed error to wrap/hide another error for instance "insufficient_quota" - if so, this is confusing.
errors messages should be as specific as possible, not wrapped into generic/incorrect messages.

**Comment by distributev** [comment]
I start to believe that a "insufficient_quota" exception is happening but on screen  it is wrapped/catched/displayed as 404 not found

**Comment by distributev** [comment]
It is exactly that once I added credit it started to work. The stack trace/error message should clearly display the root cause "insufficient_quota" (I saw this error message after I used curl to do API calls to the same endpoint to which memgpt was giving 404 "not found" errors)

I had an active subscription to chat.openai.com and thought I can use it for API calls too but it does not work like that. I had to add separate credit to platform.openai.com and now it works.

**Comment by scottmoney** [comment]
I ran into the same error.
I turned on debug more `memgpt run --debug` and found the following helpful nugget:

error = 404 Client Error: Not Found for url: https://api.openai.com/v1/chat/completions
HTTPError occurred, but unknown error message: {'message': 'The model `gpt-4` does not exist or you do not have access to it. Learn more: https://help.openai.com/en/articles/7102672-how-can-i-access-gpt-4.', 'type': 
'invalid_request_error', 'param': None, 'code': 'model_not_found'}
step() failed with an unrecognized exception: '404 Client Error: Not Found for url: https://api.openai.com/v1/chat/completions'

I changed to a different model `gpt-3.5-turbo` and it worked. 

The error that is displayed is technically correct, but not as helpful as it could be.
